core/src/main/java/io/questdb/cairo/ColumnVersionReader.java

```diff
@@ -26,43 +26,163 @@
-        resize(offset + areaSize);
+    public long getColumnNameTxn(long partitionTimestamp, int columnIndex) {
+        int versionRecordIndex = getRecordIndex(partitionTimestamp, columnIndex);
+        return versionRecordIndex > -1 ? cachedList.getQuick(versionRecordIndex + 2) : getDefaultColumnNameTxn(columnIndex);
+    }
+
+    public long getColumnNameTxnByIndex(int versionRecordIndex) {
+        return versionRecordIndex > -1 ? cachedList.getQuick(versionRecordIndex + 2) : -1L;
+    }
+
+    public long getColumnTop(long partitionTimestamp, int columnIndex) {
+        int index = getRecordIndex(partitionTimestamp, columnIndex);
+        return getColumnTopByIndex(index);
+    }
+
+    public long getColumnTopByIndex(int versionRecordIndex) {
+        return versionRecordIndex > -1 ? cachedList.getQuick(versionRecordIndex + 3) : 0L;
+    }
+
+    public long getColumnTopPartitionTimestamp(int columnIndex) {
+        int index = getRecordIndex(COL_TOP_DEFAULT_PARTITION, columnIndex);
+        // Column top place is used to store partition timestamp when column is added in COL_TOP_DEFAULT_PARTITION record
+        return index > -1 ? getColumnTopByIndex(index) : Long.MIN_VALUE;
+    }
+
+    public long getDefaultColumnNameTxn(int columnIndex) {
+        int index = getRecordIndex(COL_TOP_DEFAULT_PARTITION, columnIndex);
+        return index > -1 ? getColumnNameTxnByIndex(index) : -1L;
+    }
+
+    public int getRecordIndex(long partitionTimestamp, int columnIndex) {
+        int index = cachedList.binarySearchBlock(BLOCK_SIZE_MSB, partitionTimestamp, BinarySearch.SCAN_UP);
+        if (index > -1) {
+            final int sz = cachedList.size();
+            for (; index < sz && cachedList.getQuick(index) == partitionTimestamp; index += BLOCK_SIZE) {
+                final long thisIndex = cachedList.getQuick(index + 1);
+
+                if (thisIndex == columnIndex) {
+                    return index;
+                }
+
+                if (thisIndex > columnIndex) {
+                    break;
+                }
+            }
+        }
+        return -1;
+    }
+
+    public long getVersion() {
+        return version;
+    }
+
+    public ColumnVersionReader ofRO(FilesFacade ff, LPSZ fileName) {
+        version = -1;
+        if (this.mem == null || !ownMem) {
+            this.mem = Vm.getCMRInstance();
+        }
+        this.mem.of(ff, fileName, 0, HEADER_SIZE, MemoryTag.MMAP_TABLE_READER);
+        ownMem = true;
+        return this;
+    }
+
+    public void readSafe(MicrosecondClock microsecondClock, long spinLockTimeoutUs) {
+        final long tick = microsecondClock.getTicks();
+        while (true) {
+            long version = unsafeGetVersion();
+            if (version == this.version) {
+                return;
+            }
+            Unsafe.getUnsafe().loadFence();
+
+            final long offset;
+            final long size;
+
+            final boolean areaA = version % 2 == 0;
```

Comment:
i think we could use 
```
version & 1L
```
instead of 
```
version % 2
```

could be changed everywhere

Reply:
done: https://github.com/questdb/questdb/pull/1908/commits/606e0fdcf79288ad08e794210d1e051d3ff4ef48